core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -301,6 +301,28 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
     if (proxyUser != null && principal != null) {
       error("Only one of --proxy-user or --principal can be provided.")
     }
+
+    // Ensure heartbeat arguments is valid.
+    val executorTimeoutMs =
+      Utils.timeStringAsMs(sparkProperties.getOrElse(
+        "spark.storage.blockManagerSlaveTimeoutMs",
+        s"${Utils.timeStringAsSeconds(
+          sparkProperties.getOrElse("spark.network.timeout", "120s"))}s"))
+    val executorHeartbeatIntervalMs = Utils.timeStringAsMs(
+      sparkProperties.getOrElse("spark.executor.heartbeatInterval", "10s"))
+    val checkTimeoutIntervalMs =
+      Utils.timeStringAsSeconds(sparkProperties.getOrElse(
+        "spark.network.timeoutInterval",
+        s"${Utils.timeStringAsMs(sparkProperties.getOrElse(
+          "spark.storage.blockManagerTimeoutIntervalMs", "60s"))}ms")) * 1000
```

Comment:
the `* 1000` should be inside the bracket.

Reply:
Or, use timeStringAsMs?

Reply:
@scrowen @pgandhi999  spark.network.timeoutInterval uses seconds, and might not contains units like "30"(30 seconds), so we should use timeStringAsSeconds here to avoid interept it as 30ms.
See core/src/main/scala/org/apache/spark/HeartbeatReceiver.scala#L83 for details.

Reply:
OK, yeah that's why we need to centralize the definition of the config. This should wait until the related PR to update all these configs' declaration goes in, then can use those.